data/lib/src/datasource_impl/sort_datasource_impl.dart

```diff
@@ -61,6 +61,20 @@ class SortDataSourceImpl implements SortDataSource {
     return Future.sync(() async {
       if (listFiles is List<Document>) {
         (listFiles as List<Document>).sortFiles(sorter.orderBy, sorter.orderType);
+      } else if (listFiles is List<WorkGroupNode>) {
+        if (sorter.orderBy == OrderBy.fileSize) {
+          final listWorkGroupFolder = listFiles.whereType<WorkGroupFolder>().toList();
+          final listWorkGroupDocument = listFiles.whereType<WorkGroupDocument>().toList()
+            ..sortFiles(sorter.orderBy, sorter.orderType);
+          listFiles.clear();
+          if (sorter.orderType == OrderType.descending) {
+            (listFiles as List<WorkGroupNode>).addAll([listWorkGroupDocument, listWorkGroupFolder].expand((x) => x).toList());
+          } else {
+            (listFiles as List<WorkGroupNode>).addAll([listWorkGroupFolder, listWorkGroupDocument].expand((x) => x).toList());
+          }
```

Comment:
can you to it in functional programming? I really want we don't work on a mutable list.

Reply:
cool,  i fixed it